### ISSUE TITLE
Generate more export types such as QueryHookResult and MutationFn

### ISSUE THREAD

**Issue by github0013** [dotansimha/graphql-code-generator#1895]
**Is your feature request related to a problem? Please describe.**

There are cases where you need to pass `refetch` or `mutation` as a prop.   You currently need to dig in and pull out all types you need and then define your Props interface.

```ts
import Generated from "./generated"
import { ApolloQueryResult } from "apollo-client"

interface Props{
  refetch: (variables?: Generated.YourQueryVariables | undefined) => Promise<ApolloQueryResult<Generated.YourQuery>>
}

```

**Describe the solution you'd like**

If any of the plugins had export types for such, it would be as simple as this.

```ts
//generated.ts
export type YourQueryHookResult = QueryHookResult<
  YourQuery,
  YourQueryVariables
>

```

```ts
import Generated from "./generated"

interface Props{
  refetch: Generated.YourQueryHookResult["refetch"]

}

```

And the same goes to others like `muatation` and so on.

**Describe alternatives you've considered**

**Additional context**

**Comment by ardatan** [comment]
@dotansimha We can wait for the stable release of Apollo's own hooks API, so we can make it default in our plugin; because this kind of change will make our plugin stick to `react-apollo-hooks`.

**Comment by dotansimha** [comment]
@ardatan I think the API is pretty stable, do you think we can add it now?

**Comment by dotansimha** [comment]
Available in 1.4.0 🎉
